bot.go

```diff
@@ -127,11 +130,17 @@ func (b *Bot) SendMessage(target string, message string, sender *User) {
 	}
 }
 
-func (b *Bot) messageSender() {
+func (b *Bot) sendResponse(target, message string, sender *User) {
+	b.mutex.Lock()
+	b.handlers.Response(target, message, sender)
+	b.mutex.Unlock()
```

Comment:
This code should be: ```
b.mutex.Lock()
defer b.mutex.Unlock()
b.handlers.Response(target, message, sender)
```
This way it is exception safe.

That said, this mutex is unnecessary. The only thing that ever calls sendResponse will be a single goroutine.

Reply:
That's right, I'll fix that.